### PR TITLE
Add voting node implementation

### PR DESCRIPTION
**Pull request by Vikkingg13**

### CODE REVIEW

File: src/main/java/com/reckue/post/models/PollNode.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.List;
+
+/**
+ * Class PollNode is responsible for the voting system.
```
Comment (coungard) [inline src/main/java/com/reckue/post/models/PollNode.java:14]: Please correct author initials.

File: src/main/java/com/reckue/post/models/PollNode.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * @author Viktor Grigorirev
```
Comment (coungard) [inline src/main/java/com/reckue/post/models/PollNode.java:16]: Please correct initials.

File: src/main/java/com/reckue/post/services/realizations/PollNodeServiceRealization.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        if (node.getId() == null) {
+            throw new IllegalArgumentException("The parameter is null");
+        }
+        if (!pollNodeRepository.existsById(node.getId())) {
+            throw new ModelNotFoundException("PollNode by id " + node.getId() + " is not found");
+        }
+        PollNode savedPollNode = PollNode.builder()
+                .id(node.getId())
+                .title(node.getTitle())
+                .build();
```
Comment (coungard) [inline src/main/java/com/reckue/post/services/realizations/PollNodeServiceRealization.java:67]: Considering that there are **items** in the node, they should be here into builder too.

File: src/main/java/com/reckue/post/repositories/PollNodeRepository.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.reckue.post.repositories;
+
+import com.reckue.post.models.PollNode;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
```
Comment (camelya58) [inline src/main/java/com/reckue/post/repositories/PollNodeRepository.java:6]: Please, add javadoc

File: src/main/java/com/reckue/post/services/realizations/PollNodeServiceRealization.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            Collections.reverse(node);
+            return node;
+        }
+        return findAllBySortType(sort);
+    }
+
+    /**
+     * This method is used to sort objects by type.
+     *
+     * @param sort type of sorting: name, default - id
```
Comment (camelya58) [inline src/main/java/com/reckue/post/services/realizations/PollNodeServiceRealization.java:126]: You need to correct sorting type - tittle and add another types which you will add to PollNode.

File: src/test/java/com/reckue/post/services/realizations/NodeServiceRealizationTest.java [hunk 6]
```diff
@@ -62,6 +63,20 @@ public void create() {
         assertEquals(node, nodeService.create(node));
     }
 
+    @Test
+    public void createWithPoolNode() {
```
Comment (camelya58) [inline src/test/java/com/reckue/post/services/realizations/NodeServiceRealizationTest.java:67]: Please, correct the method name - Pool to Poll.

**Pull request merged**